test/test_interpreter.cpp

```diff
@@ -17,12 +17,52 @@
 #include "../src/xsystem.hpp"
 #include "../src/xmagics/os.hpp"
 
+#include <iostream>
+
 #include <fstream>
 #if defined(__GNUC__) && !defined(XEUS_CPP_EMSCRIPTEN_WASM_BUILD)
     #include <sys/wait.h>
     #include <unistd.h>
 #endif
 
+/**
+ * @class StreamRedirectRAII
+ * @brief A RAII class to redirect a stream to a stringstream.
+ *
+ * This class redirects the output of a given std::ostream to a std::stringstream.
+ * The original stream is restored when the object is destroyed.
+ */
```

Comment:
```suggestion
 /// A RAII class to redirect a stream to a stringstream.
 ///
 /// This class redirects the output of a given std::ostream to a std::stringstream.
 /// The original stream is restored when the object is destroyed.
```

Reply:
I've been used to this style of doxygen comments from llvm. Probably easier when building doxygen documentation one day in terms of configuration, etc. I did not find a lot of API documentation in the xeus-cpp, so maybe we should establish that style here...